scrapy/commands/startproject.py

```diff
@@ -90,36 +90,36 @@ def run(self, args, opts):
             raise UsageError()
 
         project_name = args[0]
-        project_dir = args[0]
 
         if len(args) == 2:
-            project_dir = args[1]
+            project_dir = Path(args[1])
+        else:
+            project_dir = Path(args[0])
 
-        if exists(join(project_dir, 'scrapy.cfg')):
+        if (project_dir / 'scrapy.cfg').exists():
             self.exitcode = 1
-            print(f'Error: scrapy.cfg already exists in {abspath(project_dir)}')
+            print(f'Error: scrapy.cfg already exists in {project_dir.resolve()}')
             return
 
         if not self._is_valid_name(project_name):
             self.exitcode = 1
             return
 
-        self._copytree(self.templates_dir, abspath(project_dir))
-        move(join(project_dir, 'module'), join(project_dir, project_name))
+        self._copytree(Path(self.templates_dir), project_dir.resolve())
+        move(project_dir / 'module', project_dir / project_name)
         for paths in TEMPLATES_TO_RENDER:
-            path = join(*paths)
-            tplfile = join(project_dir, string.Template(path).substitute(project_name=project_name))
-            render_templatefile(tplfile, project_name=project_name, ProjectName=string_camelcase(project_name))
+            tplfile = Path(project_dir, *(string.Template(s).substitute(project_name=project_name) for s in paths))
+            render_templatefile(str(tplfile), project_name=project_name, ProjectName=string_camelcase(project_name))
```

Comment:
It seems we can avoid calling `str(tplfile)` (here and in other tests) if the first argument of render_teplatefile would be type annotated as `Union[str, PathLike]`; it seems it won't need code changes, just annotation changes.